src/main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+)
+
+type logger interface {
+	log(format string, args ...interface{}) error
+}
+
+type checker interface {
+	run(fset *token.FileSet, logger logger) error
+}
+
+type consoleLogger struct{}
```

Comment:
I almost 99% certain that we will only need console (stdout/stderr) loggers. :)

Reply:
It was for testing purposes. I think it would be nice to have opportunity to write fake logger to write unit tests.

Reply:
I also thought about warning model which would allow writing in different formats (like json, xml, etc.)

Reply:
We may come up to some warning representation eventually.
What about using `error` for now?
When we will implement several checkers, it will be more apparent what we need and what we want from that type.